### ISSUE TITLE
Segmentation fault (Armory v0.94.0, Core v0.12.0, Debian Jessie)

### ISSUE THREAD

**Issue by lettersize** [goatpig/BitcoinArmory#31]
I am getting a "Segmentation Fault" when I run Armory (terminal output pasted at bottom of this post). Things get started just fine (Blockchain loaded, wallets synced!) but Armory crashes after a few minutes or so (it happens every time a new block is received, but I think I caught it crash once at another time). My OS is Debian Jessie, and my setup process was as follows:

* first installed Bitcoin Core v0.12.0 using the method that extracts the tar.gz and installing the binaries there into /usr/local/bin (as described here: https://bitcoin.org/en/full-node#other-linux-distributions)
* never ran bitcoind (so bitcoin.conf would not have been set up prior to Armory invoking it)
* I did run /usr/local/bin/bitcoin-qt a bunch of times until the blockchain was downloaded, before getting Armory; user that ran bitcoin-qt is the same user that built and runs Armory
* downloaded BitcoinArmory v0.94.0 from the tar.gz and successfully built from source
* I run Armory by doing `python ArmoryQt.py` in a terminal (did not install)

I wonder if setting things up again (it would be no prob to reinstall Debian etc), and just letting Armory invoke bitcoind and downloading the blockchain that way would work -- I'm happy to try that if it's got a chance of fixing the crash for me.

Terminal output from one of the crashes is below (they were all like this). Thank you!

flynn@debian:~/Downloads/BitcoinArmory-0.94.0$ python ArmoryQt.py
Gtk-Message: Failed to load module "canberra-gtk-module"
********************************************************************************
Loading Armory Engine:
   Armory Version:       0.94
   Armory Build:         None
   PyBtcWallet  Version: 1.35
Detected Operating system: Linux
   OS Variant            : ('debian', '8.3', '')
   User home-directory   : /home/flynn
   Satoshi BTC directory : /home/flynn/.bitcoin/
   Armory home dir       : /home/flynn/.armory/
   ArmoryDB directory     : /home/flynn/.armory/databases
   Armory settings file  : /home/flynn/.armory/ArmorySettings.txt
   Armory log file       : /home/flynn/.armory/armorylog.txt
   Do wallet checking    : True
-INFO  - 1459127214: (BlockUtils.cpp:924) blkfile dir: /home/flynn/.bitcoin/blocks
-INFO  - 1459127214: (BlockUtils.cpp:925) lmdb dir: /home/flynn/.armory/databases
-INFO  - 1459127214: (lmdb_wrapper.cpp:387) Opening databases...
-INFO  - 1459127214: (BlockUtils.cpp:1110) Executing: doInitialSyncOnLoad
-INFO  - 1459127214: (DatabaseBuilder.cpp:162) Reading headers from db
-INFO  - 1459127216: (DatabaseBuilder.cpp:195) Found 404592 headers in db
-INFO  - 1459127217: (DatabaseBuilder.cpp:43) updating HEADERS db
-INFO  - 1459127217: (DatabaseBuilder.cpp:378) Found next block after skipping 424402bytes
-INFO  - 1459127218: (DatabaseBuilder.cpp:223) parsed block file #477
-DEBUG - 1459127218: (Blockchain.cpp:213) Organizing chain 
-INFO  - 1459127218: (DatabaseBuilder.cpp:47) updated HEADERS db in 0.524316s
-INFO  - 1459127218: (BlockDataViewer.cpp:155) Enabling zero-conf tracking 
Segmentation fault

**Comment by achow101** [comment]
What is your OS and is it 32 or 64 bit?

If you google that error message, you could try one of the solutions suggested for other software that have caused a similar error message. From my quick search, you probably need to run this command:
`sudo apt-get install libcanberra-gtk3-module`

**Comment by lettersize** [comment]
Thanks for the suggestion, I tried the command, turns out libcanberra-gtk3-module is already installed. OS is Debian Jessie 8.3, 64 bit.

I will search for Segmentation fault errors more generally, and if anything works for me I'll post what worked here. I only searched for Segmentation faults in connection with Armory (from which I took away that data directories might be relevant, but I'm using all the default directories in any case).

**Comment by achow101** [comment]
What you want to search for are errors related to `Gtk-Message: Failed to load module "canberra-gtk-module"`as seg faults are a generic error.

**Comment by goatpig** [comment]
Are you running my builds or your own? Try to reproduce this on testnet.

**Comment by lettersize** [comment]
@achow101 Regarding `Gtk-Message: Failed to load module "canberra-gtk-module"` what I had to do was `apt-get install libcanberra-gtk*` and that eliminated that particular error message -- thanks! The `Segmentation fault` is unchanged however.

@goatpig I'm running my own build. Okay I will try to reproduce on testnet. I looked around but I'm not sure how to run on testnet when you build from source. Would I run the command `python ArmoryQt.py testnet`? And should I reconfigure Bitcoin Core to run on testnet first?

**Comment by goatpig** [comment]
1) Start Armory in offline testnet: python ArmoryQt.py --testnet --offline
2) Turn off auto bitcoind management in File -> Settings. Close Armory
3) Start BitcoinQt in testnet: bitcoin -testnet
4) Let it sync all the way
5) With BitcoinQt running, start Armory in testnet: python ArmoryQt.py --testnet

Since you are using your own build, make sure you are running off of the last commit in master. Also, run the unit tests:

1) After a successful make, browse to ./cppForSwig/gtest
2) make clean, make
3) ./CppBlockUtilsTests

**Comment by lettersize** [comment]
On testnet now, and I'm also seeing the `Segmentation fault` upon arrival of a new block. (Thank you for the step-by-step testnet instructions, very helpful.)

Below is the output I got from running the unit tests. For these tests and for the testnet run above, I built Armory from a fresh git clone of the repo this morning, which I think by default gives me the latest master commit but if I'm mistaken let me know and I'll build again.

`flynn@debian:~/Downloads/BitcoinArmory/cppForSwig/gtest$ ./CppBlockUtilsTests
Running main() from gtest_main.cc
[==========] Running 144 tests from 13 test cases.
[----------] Global test environment set-up.
[----------] 1 test from CryptoPPTest
[ RUN      ] CryptoPPTest.DetSigning
[       OK ] CryptoPPTest.DetSigning (35 ms)
[----------] 1 test from CryptoPPTest (35 ms total)

[----------] 15 tests from BinaryDataTest
[ RUN      ] BinaryDataTest.Constructor
[       OK ] BinaryDataTest.Constructor (0 ms)
[ RUN      ] BinaryDataTest.CopyFrom
[       OK ] BinaryDataTest.CopyFrom (0 ms)
[ RUN      ] BinaryDataTest.CopyTo
[       OK ] BinaryDataTest.CopyTo (0 ms)
[ RUN      ] BinaryDataTest.Fill
[       OK ] BinaryDataTest.Fill (0 ms)
[ RUN      ] BinaryDataTest.IndexOp
[       OK ] BinaryDataTest.IndexOp (0 ms)
[ RUN      ] BinaryDataTest.StartsEndsWith
[       OK ] BinaryDataTest.StartsEndsWith (0 ms)
[ RUN      ] BinaryDataTest.Append
[       OK ] BinaryDataTest.Append (0 ms)
[ RUN      ] BinaryDataTest.Inequality
[       OK ] BinaryDataTest.Inequality (0 ms)
[ RUN      ] BinaryDataTest.Equality
[       OK ] BinaryDataTest.Equality (0 ms)
[ RUN      ] BinaryDataTest.ToString
[       OK ] BinaryDataTest.ToString (0 ms)
[ RUN      ] BinaryDataTest.Endianness
[       OK ] BinaryDataTest.Endianness (0 ms)
[ RUN      ] BinaryDataTest.IntToBinData
[       OK ] BinaryDataTest.IntToBinData (0 ms)
[ RUN      ] BinaryDataTest.BinDataToInt
[       OK ] BinaryDataTest.BinDataToInt (0 ms)
[ RUN      ] BinaryDataTest.Find
[       OK ] BinaryDataTest.Find (0 ms)
[ RUN      ] BinaryDataTest.Contains
[       OK ] BinaryDataTest.Contains (0 ms)
[----------] 15 tests from BinaryDataTest (0 ms total)

[----------] 9 tests from BinaryDataRefTest
[ RUN      ] BinaryDataRefTest.Constructor
[       OK ] BinaryDataRefTest.Constructor (0 ms)
[ RUN      ] BinaryDataRefTest.PostConstruct
[       OK ] BinaryDataRefTest.PostConstruct (0 ms)
[ RUN      ] BinaryDataRefTest.CopyTo
[       OK ] BinaryDataRefTest.CopyTo (0 ms)
[ RUN      ] BinaryDataRefTest.ToString
[       OK ] BinaryDataRefTest.ToString (0 ms)
[ RUN      ] BinaryDataRefTest.Find
[       OK ] BinaryDataRefTest.Find (0 ms)
[ RUN      ] BinaryDataRefTest.Contains
[       OK ] BinaryDataRefTest.Contains (0 ms)
[ RUN      ] BinaryDataRefTest.StartsEndsWith
[       OK ] BinaryDataRefTest.StartsEndsWith (0 ms)
[ RUN      ] BinaryDataRefTest.Inequality
[       OK ] BinaryDataRefTest.Inequality (0 ms)
[ RUN      ] BinaryDataRefTest.Equality
[       OK ] BinaryDataRefTest.Equality (0 ms)
[----------] 9 tests from BinaryDataRefTest (1 ms total)

[----------] 8 tests from BitReadWriteTest
[ RUN      ] BitReadWriteTest.Writer8
[       OK ] BitReadWriteTest.Writer8 (0 ms)
[ RUN      ] BitReadWriteTest.Writer16
[       OK ] BitReadWriteTest.Writer16 (0 ms)
[ RUN      ] BitReadWriteTest.Writer32
[       OK ] BitReadWriteTest.Writer32 (0 ms)
[ RUN      ] BitReadWriteTest.Writer64
[       OK ] BitReadWriteTest.Writer64 (0 ms)
[ RUN      ] BitReadWriteTest.Reader8
[       OK ] BitReadWriteTest.Reader8 (0 ms)
[ RUN      ] BitReadWriteTest.Reader16
[       OK ] BitReadWriteTest.Reader16 (0 ms)
[ RUN      ] BitReadWriteTest.Reader32
[       OK ] BitReadWriteTest.Reader32 (0 ms)
[ RUN      ] BitReadWriteTest.Reader64
[       OK ] BitReadWriteTest.Reader64 (0 ms)
[----------] 8 tests from BitReadWriteTest (0 ms total)

[----------] 4 tests from BinaryReadWriteTest
[ RUN      ] BinaryReadWriteTest.Writer
[       OK ] BinaryReadWriteTest.Writer (0 ms)
[ RUN      ] BinaryReadWriteTest.WriterEndian
[       OK ] BinaryReadWriteTest.WriterEndian (0 ms)
[ RUN      ] BinaryReadWriteTest.Reader
[       OK ] BinaryReadWriteTest.Reader (0 ms)
[ RUN      ] BinaryReadWriteTest.ReaderEndian
[       OK ] BinaryReadWriteTest.ReaderEndian (0 ms)
[----------] 4 tests from BinaryReadWriteTest (0 ms total)

[----------] 19 tests from BtcUtilsTest
[ RUN      ] BtcUtilsTest.ReadVarInt
[       OK ] BtcUtilsTest.ReadVarInt (0 ms)
[ RUN      ] BtcUtilsTest.Num2Str
[       OK ] BtcUtilsTest.Num2Str (0 ms)
[ RUN      ] BtcUtilsTest.PackBits
[       OK ] BtcUtilsTest.PackBits (0 ms)
[ RUN      ] BtcUtilsTest.SimpleHash
[       OK ] BtcUtilsTest.SimpleHash (1 ms)
[ RUN      ] BtcUtilsTest.TxOutScriptID_Hash160
[       OK ] BtcUtilsTest.TxOutScriptID_Hash160 (0 ms)
[ RUN      ] BtcUtilsTest.TxOutScriptID_PubKey65
[       OK ] BtcUtilsTest.TxOutScriptID_PubKey65 (0 ms)
[ RUN      ] BtcUtilsTest.TxOutScriptID_PubKey33
[       OK ] BtcUtilsTest.TxOutScriptID_PubKey33 (0 ms)
[ RUN      ] BtcUtilsTest.TxOutScriptID_NonStd
[       OK ] BtcUtilsTest.TxOutScriptID_NonStd (0 ms)
[ RUN      ] BtcUtilsTest.TxOutScriptID_P2SH
[       OK ] BtcUtilsTest.TxOutScriptID_P2SH (0 ms)
[ RUN      ] BtcUtilsTest.TxOutScriptID_Multisig
[       OK ] BtcUtilsTest.TxOutScriptID_Multisig (0 ms)
[ RUN      ] BtcUtilsTest.TxOutScriptID_MultiList
[       OK ] BtcUtilsTest.TxOutScriptID_MultiList (0 ms)
[ RUN      ] BtcUtilsTest.TxInScriptID_StdUncompr
[       OK ] BtcUtilsTest.TxInScriptID_StdUncompr (0 ms)
[ RUN      ] BtcUtilsTest.TxInScriptID_StdCompr
[       OK ] BtcUtilsTest.TxInScriptID_StdCompr (0 ms)
[ RUN      ] BtcUtilsTest.TxInScriptID_Coinbase
[       OK ] BtcUtilsTest.TxInScriptID_Coinbase (0 ms)
[ RUN      ] BtcUtilsTest.TxInScriptID_SpendPubKey
[       OK ] BtcUtilsTest.TxInScriptID_SpendPubKey (1 ms)
[ RUN      ] BtcUtilsTest.TxInScriptID_SpendMultisig
[       OK ] BtcUtilsTest.TxInScriptID_SpendMultisig (0 ms)
[ RUN      ] BtcUtilsTest.TxInScriptID_SpendP2SH
[       OK ] BtcUtilsTest.TxInScriptID_SpendP2SH (0 ms)
[ RUN      ] BtcUtilsTest.BitsToDifficulty
[       OK ] BtcUtilsTest.BitsToDifficulty (0 ms)
[ RUN      ] BtcUtilsTest.ScriptToOpCodes
[       OK ] BtcUtilsTest.ScriptToOpCodes (0 ms)
[----------] 19 tests from BtcUtilsTest (2 ms total)

[----------] 10 tests from BlockObjTest
[ RUN      ] BlockObjTest.HeaderNoInit
[       OK ] BlockObjTest.HeaderNoInit (0 ms)
[ RUN      ] BlockObjTest.HeaderUnserialize
[       OK ] BlockObjTest.HeaderUnserialize (0 ms)
[ RUN      ] BlockObjTest.HeaderProperties
[       OK ] BlockObjTest.HeaderProperties (0 ms)
[ RUN      ] BlockObjTest.OutPointProperties
[       OK ] BlockObjTest.OutPointProperties (1 ms)
[ RUN      ] BlockObjTest.OutPointSerialize
[       OK ] BlockObjTest.OutPointSerialize (0 ms)
[ RUN      ] BlockObjTest.TxInNoInit
[       OK ] BlockObjTest.TxInNoInit (0 ms)
[ RUN      ] BlockObjTest.TxInUnserialize
[       OK ] BlockObjTest.TxInUnserialize (0 ms)
[ RUN      ] BlockObjTest.TxOutUnserialize
[       OK ] BlockObjTest.TxOutUnserialize (0 ms)
[ RUN      ] BlockObjTest.TxNoInit
[       OK ] BlockObjTest.TxNoInit (0 ms)
[ RUN      ] BlockObjTest.TxUnserialize
[       OK ] BlockObjTest.TxUnserialize (2 ms)
[----------] 10 tests from BlockObjTest (3 ms total)

[----------] 39 tests from StoredBlockObjTest
[ RUN      ] StoredBlockObjTest.StoredObjNoInit
[       OK ] StoredBlockObjTest.StoredObjNoInit (0 ms)
[ RUN      ] StoredBlockObjTest.GetDBKeys
[       OK ] StoredBlockObjTest.GetDBKeys (0 ms)
[ RUN      ] StoredBlockObjTest.LengthUnfrag
[       OK ] StoredBlockObjTest.LengthUnfrag (0 ms)
[ RUN      ] StoredBlockObjTest.LengthFragged
[       OK ] StoredBlockObjTest.LengthFragged (0 ms)
[ RUN      ] StoredBlockObjTest.BlkDataKeys
[       OK ] StoredBlockObjTest.BlkDataKeys (0 ms)
[ RUN      ] StoredBlockObjTest.ReadBlkKeyData
[       OK ] StoredBlockObjTest.ReadBlkKeyData (1 ms)
[ RUN      ] StoredBlockObjTest.SHeaderUnserialize
[       OK ] StoredBlockObjTest.SHeaderUnserialize (0 ms)
[ RUN      ] StoredBlockObjTest.SHeaderDBSerFull_H
[       OK ] StoredBlockObjTest.SHeaderDBSerFull_H (0 ms)
[ RUN      ] StoredBlockObjTest.SHeaderDBSerFull_B1
[       OK ] StoredBlockObjTest.SHeaderDBSerFull_B1 (0 ms)
[ RUN      ] StoredBlockObjTest.SHeaderDBSerFull_B2
[       OK ] StoredBlockObjTest.SHeaderDBSerFull_B2 (0 ms)
[ RUN      ] StoredBlockObjTest.SHeaderDBSerFull_B3
[       OK ] StoredBlockObjTest.SHeaderDBSerFull_B3 (0 ms)
[ RUN      ] StoredBlockObjTest.SHeaderDBUnserFull_H
[       OK ] StoredBlockObjTest.SHeaderDBUnserFull_H (1 ms)
[ RUN      ] StoredBlockObjTest.SHeaderDBUnserFull_B1
[       OK ] StoredBlockObjTest.SHeaderDBUnserFull_B1 (0 ms)
[ RUN      ] StoredBlockObjTest.SHeaderDBUnserFull_B2
[       OK ] StoredBlockObjTest.SHeaderDBUnserFull_B2 (0 ms)
[ RUN      ] StoredBlockObjTest.SHeaderDBUnserFull_B3
[       OK ] StoredBlockObjTest.SHeaderDBUnserFull_B3 (0 ms)
[ RUN      ] StoredBlockObjTest.STxUnserUnfrag
[       OK ] StoredBlockObjTest.STxUnserUnfrag (0 ms)
[ RUN      ] StoredBlockObjTest.STxUnserFragged
[       OK ] StoredBlockObjTest.STxUnserFragged (0 ms)
[ RUN      ] StoredBlockObjTest.STxReconstruct
[       OK ] StoredBlockObjTest.STxReconstruct (0 ms)
[ RUN      ] StoredBlockObjTest.STxSerUnfragToFrag
[       OK ] StoredBlockObjTest.STxSerUnfragToFrag (0 ms)
[ RUN      ] StoredBlockObjTest.STxSerDBValue_1
[       OK ] StoredBlockObjTest.STxSerDBValue_1 (0 ms)
[ RUN      ] StoredBlockObjTest.STxUnserDBValue_1
[       OK ] StoredBlockObjTest.STxUnserDBValue_1 (0 ms)
[ RUN      ] StoredBlockObjTest.STxUnserDBValue_2
[       OK ] StoredBlockObjTest.STxUnserDBValue_2 (0 ms)
[ RUN      ] StoredBlockObjTest.STxOutUnserialize
[       OK ] StoredBlockObjTest.STxOutUnserialize (0 ms)
[ RUN      ] StoredBlockObjTest.STxOutSerDBValue_1
[       OK ] StoredBlockObjTest.STxOutSerDBValue_1 (0 ms)
[ RUN      ] StoredBlockObjTest.STxOutSerDBValue_2
[       OK ] StoredBlockObjTest.STxOutSerDBValue_2 (0 ms)
[ RUN      ] StoredBlockObjTest.STxOutSerDBValue_3
[       OK ] StoredBlockObjTest.STxOutSerDBValue_3 (0 ms)
[ RUN      ] StoredBlockObjTest.STxOutUnserDBValue_1
[       OK ] StoredBlockObjTest.STxOutUnserDBValue_1 (0 ms)
[ RUN      ] StoredBlockObjTest.STxOutUnserDBValue_2
[       OK ] StoredBlockObjTest.STxOutUnserDBValue_2 (0 ms)
[ RUN      ] StoredBlockObjTest.STxOutUnserDBValue_3
[       OK ] StoredBlockObjTest.STxOutUnserDBValue_3 (1 ms)
[ RUN      ] StoredBlockObjTest.SHeaderFullBlock
[       OK ] StoredBlockObjTest.SHeaderFullBlock (0 ms)
[ RUN      ] StoredBlockObjTest.SUndoDataSer
[       OK ] StoredBlockObjTest.SUndoDataSer (0 ms)
[ RUN      ] StoredBlockObjTest.SUndoDataUnser
[       OK ] StoredBlockObjTest.SUndoDataUnser (1 ms)
[ RUN      ] StoredBlockObjTest.STxHintsSer
[       OK ] StoredBlockObjTest.STxHintsSer (0 ms)
[ RUN      ] StoredBlockObjTest.STxHintsReorder
[       OK ] StoredBlockObjTest.STxHintsReorder (0 ms)
[ RUN      ] StoredBlockObjTest.STxHintsUnser
[       OK ] StoredBlockObjTest.STxHintsUnser (0 ms)
[ RUN      ] StoredBlockObjTest.SHeadHgtListSer
[       OK ] StoredBlockObjTest.SHeadHgtListSer (0 ms)
[ RUN      ] StoredBlockObjTest.SHeadHgtListUnser
[       OK ] StoredBlockObjTest.SHeadHgtListUnser (0 ms)
[ RUN      ] StoredBlockObjTest.SScriptHistorySer
[       OK ] StoredBlockObjTest.SScriptHistorySer (1 ms)
[ RUN      ] StoredBlockObjTest.SScriptHistoryUnser
[       OK ] StoredBlockObjTest.SScriptHistoryUnser (0 ms)
[----------] 39 tests from StoredBlockObjTest (7 ms total)

[----------] 8 tests from LMDBTest
[ RUN      ] LMDBTest.OpenClose
[       OK ] LMDBTest.OpenClose (6 ms)
[ RUN      ] LMDBTest.OpenCloseOpenNominal
[       OK ] LMDBTest.OpenCloseOpenNominal (12 ms)
[ RUN      ] LMDBTest.PutGetDelete
[       OK ] LMDBTest.PutGetDelete (5 ms)
[ RUN      ] LMDBTest.STxOutPutGet
[       OK ] LMDBTest.STxOutPutGet (5 ms)
[ RUN      ] LMDBTest.PutGetBareHeader
[       OK ] LMDBTest.PutGetBareHeader (5 ms)
[ RUN      ] LMDBTest.PutGetStoredTxHints
[       OK ] LMDBTest.PutGetStoredTxHints (4 ms)
[ RUN      ] LMDBTest.PutGetStoredScriptHistory
[       OK ] LMDBTest.PutGetStoredScriptHistory (6 ms)
[ RUN      ] LMDBTest.HeaderDump
[       OK ] LMDBTest.HeaderDump (2 ms)
[----------] 8 tests from LMDBTest (46 ms total)

[----------] 2 tests from TxRefTest
[ RUN      ] TxRefTest.TxRefNoInit
[       OK ] TxRefTest.TxRefNoInit (0 ms)
[ RUN      ] TxRefTest.TxRefKeyParts
[       OK ] TxRefTest.TxRefKeyParts (0 ms)
[----------] 2 tests from TxRefTest (0 ms total)

[----------] 20 tests from BlockUtilsBare
[ RUN      ] BlockUtilsBare.BuildNoRegisterWlt
[       OK ] BlockUtilsBare.BuildNoRegisterWlt (12 ms)
[ RUN      ] BlockUtilsBare.Load5Blocks
[       OK ] BlockUtilsBare.Load5Blocks (33 ms)
[ RUN      ] BlockUtilsBare.Load5Blocks_DamagedBlkFile
[       OK ] BlockUtilsBare.Load5Blocks_DamagedBlkFile (22 ms)
[ RUN      ] BlockUtilsBare.Load4Blocks_Plus2
[       OK ] BlockUtilsBare.Load4Blocks_Plus2 (29 ms)
[ RUN      ] BlockUtilsBare.Load4Blocks_ReloadBDM_ZC_Plus2
[       OK ] BlockUtilsBare.Load4Blocks_ReloadBDM_ZC_Plus2 (38 ms)
[ RUN      ] BlockUtilsBare.Load3Blocks_ZC_Plus3_TestLedgers
[       OK ] BlockUtilsBare.Load3Blocks_ZC_Plus3_TestLedgers (42 ms)
[ RUN      ] BlockUtilsBare.Load3Blocks_ZCchain
[       OK ] BlockUtilsBare.Load3Blocks_ZCchain (35 ms)
[ RUN      ] BlockUtilsBare.Load3Blocks_RBF
[       OK ] BlockUtilsBare.Load3Blocks_RBF (29 ms)
[ RUN      ] BlockUtilsBare.Load5Blocks_FullReorg
[       OK ] BlockUtilsBare.Load5Blocks_FullReorg (29 ms)
[ RUN      ] BlockUtilsBare.Load5Blocks_DoubleReorg
[       OK ] BlockUtilsBare.Load5Blocks_DoubleReorg (32 ms)
[ RUN      ] BlockUtilsBare.Load5Blocks_ReloadBDM_Reorg
[       OK ] BlockUtilsBare.Load5Blocks_ReloadBDM_Reorg (33 ms)
[ RUN      ] BlockUtilsBare.Load5Blocks_ReloadBDM_Reorg_DontTrigger
[       OK ] BlockUtilsBare.Load5Blocks_ReloadBDM_Reorg_DontTrigger (39 ms)
[ RUN      ] BlockUtilsBare.CorruptedBlock
[       OK ] BlockUtilsBare.CorruptedBlock (31 ms)
[ RUN      ] BlockUtilsBare.Load5Blocks_RescanOps
[       OK ] BlockUtilsBare.Load5Blocks_RescanOps (42 ms)
[ RUN      ] BlockUtilsBare.Load5Blocks_RescanEmptyDB
[       OK ] BlockUtilsBare.Load5Blocks_RescanEmptyDB (27 ms)
[ RUN      ] BlockUtilsBare.Load5Blocks_RebuildEmptyDB
[       OK ] BlockUtilsBare.Load5Blocks_RebuildEmptyDB (29 ms)
[ RUN      ] BlockUtilsBare.Load5Blocks_ForceFullRewhatever
[       OK ] BlockUtilsBare.Load5Blocks_ForceFullRewhatever (43 ms)
[ RUN      ] BlockUtilsBare.Load5Blocks_ScanWhatIsNeeded
[       OK ] BlockUtilsBare.Load5Blocks_ScanWhatIsNeeded (33 ms)
[ RUN      ] BlockUtilsBare.Load5Blocks_GetUtxos
[       OK ] BlockUtilsBare.Load5Blocks_GetUtxos (29 ms)
[ RUN      ] BlockUtilsBare.Load4Blocks_ZC_GetUtxos
[       OK ] BlockUtilsBare.Load4Blocks_ZC_GetUtxos (27 ms)
[----------] 20 tests from BlockUtilsBare (635 ms total)

[----------] 6 tests from BlockDir
[ RUN      ] BlockDir.HeadersFirst
[       OK ] BlockDir.HeadersFirst (17 ms)
[ RUN      ] BlockDir.HeadersFirstUpdate
[       OK ] BlockDir.HeadersFirstUpdate (18 ms)
[ RUN      ] BlockDir.HeadersFirstReorg
[       OK ] BlockDir.HeadersFirstReorg (20 ms)
[ RUN      ] BlockDir.HeadersFirstUpdateTwice
[       OK ] BlockDir.HeadersFirstUpdateTwice (19 ms)
[ RUN      ] BlockDir.BlockFileSplit
[       OK ] BlockDir.BlockFileSplit (18 ms)
[ RUN      ] BlockDir.BlockFileSplitUpdate
[       OK ] BlockDir.BlockFileSplitUpdate (17 ms)
[----------] 6 tests from BlockDir (110 ms total)

[----------] 3 tests from TestCryptoECDSA
[ RUN      ] TestCryptoECDSA.VerifySECP256K1Point
[       OK ] TestCryptoECDSA.VerifySECP256K1Point (5 ms)
[ RUN      ] TestCryptoECDSA.SECP256K1MultScalars
[       OK ] TestCryptoECDSA.SECP256K1MultScalars (0 ms)
[ RUN      ] TestCryptoECDSA.VerifyPubKeyValidity
[       OK ] TestCryptoECDSA.VerifyPubKeyValidity (19 ms)
[----------] 3 tests from TestCryptoECDSA (24 ms total)

[----------] Global test environment tear-down
[==========] 144 tests from 13 test cases ran. (863 ms total)
[  PASSED  ] 144 tests.

  YOU HAVE 36 DISABLED TESTS

Segmentation fault`

**Comment by goatpig** [comment]
Will try to reproduce on my Debian 7 VM and get back to you.

**Comment by goatpig** [comment]
I cannot reproduce this error so I will need you to help me with the debugging.

Let's figure out the basis first (do all these test on testnet for the sake of speed)

1) Do you get any seg faults during the initial scan?
2) Do you get 100% seg faults on new blocks?
3) Do you get seg faults on new blocks after an initial scan (delete your DB folder, let Armory build up and let it run till a new block hits the testnet)

4) Does the wallet consistency scan complete before the segfaults? Try disabling it (comment out these 2 lines: https://github.com/goatpig/BitcoinArmory/blob/master/ArmoryQt.py#L819)

**Comment by lettersize** [comment]
I think we found my problem -- I had always been skipping the initial wallet creation, intending to import a watching only wallet later from offline, when my online Armory setup was ready. Didn't occur to me this could cause an issue.

When I created a wallet to see about the consistency scan for step 4), the segfaults did not occur. Armory is learning about new blocks with no segfault now, both on testnet and on mainnet.

For the sake of completeness here's what I found for 1)-4):
--WITHOUT WALLET
1) No
2) Yes, crash happened with arrival of every new block, and never had any crashes between blocks.
3) Yes
--WITH WALLET
4) Yes the consistency scan completes, and the segfaults do not occur. Everything's fine with the scan disabled too.

Thanks for the help! Happy to provide any more info if wanted.
